closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -666,263 +719,49 @@ private static final class RestrictedSecurityProperties {
         // The map is keyed by provider name.
         private final Map<String, Constraint[]> providerConstraints;
 
-        private final String profileID;
-
-        // The java.security properties.
-        private final Properties securityProps;
-
-        /**
-         *
-         * @param id    the restricted security custom profile ID
-         * @param props the java.security properties
-         * @param trace the user security trace
-         * @param audit the user security audit
-         * @param help  the user security help
-         */
-        private RestrictedSecurityProperties(String id, Properties props) {
-            Objects.requireNonNull(props);
-
-            profileID = id;
-            securityProps = props;
+        private RestrictedSecurityProperties(String profileID, String descName,
+                                             boolean descIsDefault, boolean descIsFIPS,
+                                             String descNumber, String descPolicy,
+                                             String descSunsetDate,
+                                             String jdkTlsDisabledNamedCurves,
+                                             String jdkTlsDisabledAlgorithms,
+                                             String jdkTlsEphemeralDHKeySize,
+                                             String jdkTlsLegacyAlgorithms,
+                                             String jdkCertpathDisabledAlgorithms,
+                                             String jdkSecurityLegacyAlgorithms,
+                                             String keyStoreType,
+                                             String keyStore,
+                                             String jdkSecureRandomProvider,
+                                             String jdkSecureRandomAlgorithm,
+                                             String jdkFipsMode) {
```

Comment:
Rather than extract all those values at the call site (with the potential to get the order wrong), suggest passing the `ProfileParser` and allowing this constructor to query the values here. `ProfileParser` would simply
```java
        private RestrictedSecurityProperties getProperties() {
            return new RestrictedSecurityProperties(profileID, this);
        }
```

Reply:
Changed to suggestion.